include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,533 @@
+        vk::resultCheck( vk::Result{ result }, "Mman: allocator creation error." );
+    } // Mman
+
+    ~Mman() { vmaDestroyAllocator( m_vma ); }
+
+    Mman( const Mman& ) = delete;
+    Mman( Mman&& ) = delete;
+    Mman& operator=( const Mman& ) = delete;
+    Mman& operator=( Mman&& ) = delete;
+
+    vk::Buffer create( const vk::BufferCreateInfo& create_info )
+    {
+        VkBuffer buffer{};
+        VmaAllocation allocation{};
+
+        VkResult result = vmaCreateBuffer(
+            m_vma,
+            &static_cast<const VkBufferCreateInfo&>( create_info ),
+            &k_buffer_alloc_create_info,
+            &buffer,
+            &allocation,
+            nullptr );
+
+        vk::resultCheck( vk::Result{ result }, "Mman: buffer allocation error." );
+
+        vk::Buffer buffer_hpp{ buffer };
+        BufferInfo buffer_info{ allocation, create_info.size };
+
+        addInfo( buffer_hpp, buffer_info );
+
+        return buffer_hpp;
+    } // create
+
+    vk::Image create( const vk::ImageCreateInfo& create_info )
+    {
+        VkImage image{};
+        VmaAllocation allocation{};
+
+        VkResult result = vmaCreateImage(
+            m_vma,
+            &static_cast<const VkImageCreateInfo&>( create_info ),
+            &k_image_alloc_create_info,
+            &image,
+            &allocation,
+            nullptr );
+
+        vk::resultCheck( vk::Result{ result }, "Mman: image allocation error." );
+
+        vk::Image image_hpp{ image };
+        ImageInfo image_info{ allocation, create_info.format, create_info.initialLayout, create_info.extent };
+
+        addInfo( image_hpp, image_info );
+
+        return image_hpp;
+    } // create
+
+    void destroy( vk::Buffer buffer )
+    {
+        vmaDestroyBuffer( m_vma, buffer, *findAllocation( buffer ) );
+        clearInfo( buffer );
+    } // destroy
+
+    void destroy( vk::Image image )
+    {
+        vmaDestroyImage( m_vma, image, *findAllocation( image ) );
+        clearInfo( image );
+    } // destroy
+
+    uint8_t* map( vk::Buffer buffer )
+    {
+        uint8_t* mapped = nullptr;
+
+        VkResult result = vmaMapMemory( m_vma, *findAllocation( buffer ), reinterpret_cast<void**>( &mapped ) );
+        vk::resultCheck( vk::Result{ result }, "Mman: buffer mapping error." );
+
+        return mapped;
+    } // map
+
+    void unmap( vk::Buffer buffer ) { vmaUnmapMemory( m_vma, *findAllocation( buffer ) ); }
+
+    void flush( vk::Buffer buffer )
+    {
+        VkResult result = vmaFlushAllocation( m_vma, *findAllocation( buffer ), 0, VK_WHOLE_SIZE );
+        vk::resultCheck( vk::Result{ result }, "Mman: buffer flushing error." );
+    } // flush
+
+    void copy(
+        vk::Buffer src_buffer,
+        vk::Buffer dst_buffer,
+        vk::DeviceSize src_offset,
+        vk::DeviceSize dst_offset,
+        vk::DeviceSize size )
+    {
+        vk::BufferCopy region{ src_offset, dst_offset, size };
+
+        createCommand().submitAndWait(
```

Comment:
One-Time submit buffer can be reused multiple times, instead of being created each time

Reply:
How it should be done?